graylog2-web-interface/src/components/sidecars/configuration-forms/ConfigurationVariablesHelper.jsx

```diff
@@ -0,0 +1,151 @@
+import { Spinner } from 'components/common';
+import { BootstrapModalConfirm } from 'components/bootstrap';
+import BootstrapModalWrapper from 'components/bootstrap/BootstrapModalWrapper';
+import CombinedProvider from 'injection/CombinedProvider';
+import EditConfigurationVariableModal from './EditConfigurationVariableModal';
+import ConfigurationHelperStyle from './ConfigurationHelper.css';
+
+const { ConfigurationVariableActions } = CombinedProvider.get('ConfigurationVariable');
+
+class ConfigurationVariablesHelper extends React.Component {
+  state = {
+    configurationVariables: [],
+    errorModalContent: {},
+  };
+
+  componentDidMount() {
+    this._reloadConfiguration();
+  }
+
+  _reloadConfiguration = () => {
+    ConfigurationVariableActions.all()
+      .then((configurationVariables) => {
+        this.setState({ configurationVariables: configurationVariables });
+      });
+  };
+
+  _getId = (idName, index) => {
+    const idIndex = index !== undefined ? `. ${index}` : '';
+    return idName + idIndex;
+  };
+
+  _renderConfigList = (configurations) => {
+    return (<ul className={ConfigurationHelperStyle.ulStyle}>
+      {configurations.map(conf => <li key={conf.id}><a href={conf.id}>{conf.name}</a></li>)}
+    </ul>);
+  };
+
+  _configurationVariableListFormatter = () => {
+    const variableRows = [];
+
+    Object.values(this.state.configurationVariables).forEach((configVar) => {
+      let deleteConfirmationModal;
+      const _handleDeleteCheck = () => {
+        ConfigurationVariableActions.getConfigurations(configVar).then((response) => {
+          // Variable still in use: Report error
+          if (response.length > 0) {
+            this.setState({ errorModalContent: this._renderConfigList(response) });
+            this.errorModal.open();
+          // Not in use, ask for confirmation
+          } else {
+            deleteConfirmationModal.open();
+          }
+        });
+      };
+
+      const _handleDeleteConfirm = () => {
+        ConfigurationVariableActions.delete(configVar)
+          .then(() => this._onSuccessfulUpdate());
+      };
+
+      const escapedName = `\${${configVar.name}}`;
+      variableRows.push(
+        <tr key={this._getId(configVar.id)}>
+          <td><code>{escapedName}</code></td>
+          <td>{configVar.description}</td>
+          <td style={{ textAlign: 'right' }}>
+            <BootstrapModalConfirm ref={(c) => { deleteConfirmationModal = c; }}
+                                   title="Delete Configuration Variable?"
+                                   onConfirm={_handleDeleteConfirm}
+                                   onCancel={() => {}}>
+              <p>Are you sure you want to remove the configuration variable <strong>{configVar.name}</strong>?</p>
+            </BootstrapModalConfirm>
+            <Button bsStyle="primary" bsSize="xsmall" onClick={_handleDeleteCheck}>
+              Delete
+            </Button>
```

Comment:
The delete and edit button have no space between them. You could either set a `margin` in one of them, or add a `&nbsp;` to avoid that.